src/usePerpsCommitOrder.ts

```diff
@@ -0,0 +1,133 @@
+  const { data: PerpsMarketProxyContract } = useImportContract('PerpsMarketProxy');
+  const { data: MulticallContract } = useImportContract('Multicall');
+  const { data: PythERC7412WrapperContract } = useImportContract('PythERC7412Wrapper');
+  const errorParser = useErrorParser();
+
+  return useMutation({
+    retry: false,
+    mutationFn: async (sizeDelta: ethers.BigNumber) => {
+      if (
+        !(
+          chainId &&
+          perpsAccountId &&
+          settlementStrategyId &&
+          priceIds &&
+          PerpsMarketProxyContract?.address &&
+          MulticallContract?.address &&
+          PythERC7412WrapperContract?.address &&
+          walletAddress &&
+          feedId &&
+          provider
+        )
+      ) {
+        throw 'OMFG';
+      }
+
+      if (sizeDelta.lte(0)) {
+        throw new Error('Amount required');
+      }
+
+      const availableMargin = await fetchPerpsGetAvailableMargin({
+        provider,
+        perpsAccountId,
+        PerpsMarketProxyContract,
+      });
+
+      if (availableMargin.lt(sizeDelta)) {
+        throw new Error('Not enough available margin');
+      }
+
+      const totalCollateralValue = await fetchPerpsTotalCollateralValue({
+        provider,
+        PerpsMarketProxyContract,
+        accountId: perpsAccountId,
+      });
+
+      if (totalCollateralValue.lt(sizeDelta)) {
+        throw new Error('Total collateral value is less than the size delta');
+      }
+
+      const pythPrice = await getPythPrice({ feedId });
+
+      const orderCommitmentArgs = {
+        marketId,
+        accountId: perpsAccountId,
+        sizeDelta,
+        settlementStrategyId,
+        acceptablePrice: ethers.utils.parseEther(Math.floor(pythPrice * (sizeDelta.gt(0) ? 1.05 : 0.95)).toString()),
+        referrer: ethers.constants.AddressZero,
+        trackingCode: ethers.utils.formatBytes32String('VD'),
+      };
+
+      const freshPriceUpdateTxn = await fetchPriceUpdateTxn({
+        provider,
+        MulticallContract,
+        PythERC7412WrapperContract,
+        priceIds,
+      });
+      console.log('freshPriceUpdateTxn', freshPriceUpdateTxn);
+
+      if (freshPriceUpdateTxn.value) {
+        console.log('-> fetchPerpsCommitOrderWithPriceUpdate');
+        await fetchPerpsCommitOrderWithPriceUpdate({
+          walletAddress,
+          provider,
+          PerpsMarketProxyContract,
+          MulticallContract,
+          orderCommitmentArgs,
+          priceUpdateTxn: freshPriceUpdateTxn,
+        });
+      } else {
+        console.log('-> fetchPerpsCommitOrder');
+        await fetchPerpsCommitOrder({
+          walletAddress,
+          provider,
+          PerpsMarketProxyContract,
+          orderCommitmentArgs,
+        });
+      }
+
+      return { priceUpdated: true };
+    },
+    throwOnError: (error) => {
+      // TODO: show toast
+      errorParser(error);
+      return false;
+    },
+    onSuccess: ({ priceUpdated }) => {
+      onSuccess({ priceUpdated });
```

Comment:
clear all the query caches here and then call `onSuccess()` (same for other places)

Reply:
update query caches, moved to `onSuccess()` callbacks of sdk hooks